src/Symfony/Component/Routing/RouteCompiler.php

```diff
@@ -223,14 +223,39 @@ private static function compilePattern(Route $route, $pattern, $isHost)
         }
 
         return array(
-            'staticPrefix' => 'text' === $tokens[0][0] ? $tokens[0][1] : '',
+            'staticPrefix' => static::determineStaticPrefix($route, $tokens),
```

Comment:
access to private APIs MUST use ``self``, not ``static``. Otherwise it breaks when using inheritance as it will use ``ChildClass::determineStaticPrefix``, which is forbidden by the visibility.